libraries/chain/database.cpp

```diff
@@ -2252,15 +2252,93 @@ namespace golos { namespace chain {
+
+                    // Also we need to distibute auction_window reward to voters from second set.
+                    // So we need to calculate parts of reward for 3 sets:
+                    //       - votes_in_auction_window_reward
+                    //       - votes_after_auction_window_reward
+                    //       - auction_window_reward 
+
+                    if (has_hardfork(STEEMIT_HARDFORK_0_19__898) && 
+                        c.auction_window_reward_destination == protocol::to_curators
+                    ) {
+                        // separate votes
+                        const auto &cvlupdidx = get_index<comment_vote_index>().indices().get<by_vote_last_update>();
+                        auto itr = cvlupdidx.lower_bound(boost::make_tuple(0, c.created));
+                        auto itr_after_auw = cvlupdidx.lower_bound( // auw -- auctcion window 
+                            boost::make_tuple(0, c.created + c.auction_window_size)
+                        );
+
+                        while (itr != itr_after_auw && itr->comment == c.id) {
+                            uint128_t weight(itr->weight);
+                            if (itr->last_update >= c.created &&
+                                itr->last_update < auw_time &&
+                                weight > 0
+                            ) {
+                                votes_in_auction_window_weight += weight;
+                            }
+                        }
+
+                        votes_after_auction_window_weight = total_weight - 
+                            votes_in_auction_window_weight - c.auction_window_weight;
+
+                        votes_in_auction_window_reward = ((max_rewards.value * votes_in_auction_window_weight) /
+                                      total_weight).to_uint64();
+
+                        auction_window_reward = ((max_rewards.value * c.auction_window_weight) /
+                                      total_weight).to_uint64();
+
+                        votes_after_auction_window_reward = ((max_rewards.value * votes_after_auction_window_weight) /
+                                      total_weight).to_uint64();
+                    }
+
                     const auto &cvidx = get_index<comment_vote_index>().indices().get<by_comment_weight_voter>();
                     auto itr = cvidx.lower_bound(c.id);
                     while (itr != cvidx.end() && itr->comment == c.id) {
                         uint128_t weight(itr->weight);
-                        auto claim = ((max_rewards.value * weight) /
+
+                        uint64_t claim;
+                        // to_curators case
+                        if (has_hardfork(STEEMIT_HARDFORK_0_19__898) && 
+                            c.auction_window_reward_destination == protocol::to_curators) {
+                            if (itr->last_update >= c.created && itr->last_update < auw_time) {
```

Comment:
The check `itr->last_update >= c.created` for what?

Reply:
It's extra check. Removed it